File: Fundamentals of C programming/step06/6.1.08.c
/*

Напишите программу считающую произведение цифр заданного kk-значного числа XX.

Входные данные: Два целых числа.
Первое число kk -- количество цифр в числе XX. 1≤k≤41≤k≤4
Второе число XX. Некоторое kk-значное число.

Выходные данные:
Целое число. Произведение цифр числа XX.

Sample Input:

2 48

Sample Output:

32

*/

#include <stdio.h>

int main() {
    int k,    x;
    scanf("%d    %d",   &k,    &x);
    switch(k){
        case    1:    printf("%d\n",    x);    break;
        case    2:    printf("%d\n",    (x/10)*(x%10));    break;
        case    3:    printf("%d\n",    (x/100)*((x/10)%10)*(x%10));    break;
        case    4:    printf("%d\n",    (x/1000)*((x/100)%10)*((x/10)%10)*(x%10));    break;
        default:    printf("ERROR!\n");    break;    
    }
  return 0;
}

File: Good, good Python/step 06/6.01/6.1.08.py
'''

Подвиг 10. Тестовый веб-сервер возвращает HTML-страницы по URL-адресам (строкам). 
На вход программы поступают различные URL-адреса. Если адрес пришел впервые, то на экране отобразить строку (без кавычек):

"HTML-страница для адреса <URL-адрес>"

Если адрес приходит повторно, то следует взять строку "HTML-страница для адреса <URL-адрес>" из словаря и вывести на экран сообщение (без кавычек):

"Взято из кэша: HTML-страница для адреса <URL-адрес>"

Сообщения выводить каждое с новой строки.

P. S. Для считывания списка целиком в программе уже записаны начальные строчки.

Sample Input:

ustanovka-i-zapusk-yazyka
ustanovka-i-poryadok-raboty-pycharm
peremennyye-operator-prisvaivaniya-tipy-dannykh
arifmeticheskiye-operatsii
ustanovka-i-poryadok-raboty-pycharm

Sample Output:

HTML-страница для адреса ustanovka-i-zapusk-yazyka
HTML-страница для адреса ustanovka-i-poryadok-raboty-pycharm
HTML-страница для адреса peremennyye-operator-prisvaivaniya-tipy-dannykh
HTML-страница для адреса arifmeticheskiye-operatsii
Взято из кэша: HTML-страница для адреса ustanovka-i-poryadok-raboty-pycharm

'''

import sys

# считывание списка из входного потока
lst_in = list(map(str.strip, sys.stdin.readlines()))

# здесь продолжайте программу (используйте список lst_in)

x = []
for i in lst_in:
    if i in x:
        print(f'Взято из кэша: HTML-страница для адреса {i}')
    else:
        print(f'HTML-страница для адреса {i}')
    x.append(i) 
